File: backend/spring-boot/src/main/java/com/runhwani/runmate/service/HistoryService.java
package com.runhwani.runmate.service;

import com.runhwani.runmate.dto.response.history.HistoryListResponse;

public interface HistoryService {
    /**
     * 사용자의 달리기 기록 목록을 페이징하여 조회
     * @param page 페이지 번호 (1부터 시작)
     * @param size 페이지 크기
     * @return 페이징된 달리기 기록 목록
     */
    HistoryListResponse getHistoryList(int page, int size);
} 

File: RunMate/data/src/main/java/com/D107/runmate/data/remote/api/HistoryService.kt
package com.D107.runmate.data.remote.api

import com.D107.runmate.data.remote.common.ApiResponse
import com.D107.runmate.data.remote.common.ServerResponse
import com.D107.runmate.data.remote.response.history.HistoryDetailResponse
import com.D107.runmate.data.remote.response.history.HistoryResponse
import com.D107.runmate.data.remote.response.history.UserHistoryDetailResponse
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface HistoryService {
    @GET("histories")
    suspend fun getHistoryList(@Query("size") size: Int): Response<ServerResponse<HistoryResponse>>

    @GET("histories/{historyId}")
    suspend fun getHistoryDetail(@Path("historyId") historyId: String): Response<ServerResponse<HistoryDetailResponse>>

    @GET("histories/{historyId}/users/{userId}")
    suspend fun getHistoryDetailByUserId(@Path("historyId") historyId: String, @Path("userId") userId: String): Response<ServerResponse<UserHistoryDetailResponse>>
}